Sources/core/objects/Movable.h

```diff
@@ -34,7 +34,7 @@ class IMovable : public IOnMapObject
     void processMove();//for move
     virtual bool TryMove(Dir direct);
     bool CheckMoveTime();
-    bool CheckPassable();
+    virtual bool CheckPassable();
     bool MainMove();
```

Comment:
Now forces are handled by force handler. `ProcessForce` function or so. Now it is kinda primitive system, but later it will be impoved. So that function/system exists, but it is not suitable for projectiles now.
So for now (and even later) it is better to remove projectiles in `CheckPassable` function, because projectiles not always should be deleted on zero speed (e.g. boomerang).